engine/src/known_types.rs

```diff
@@ -217,28 +222,12 @@ impl TypeDatabase {
         pod_safety.into_iter()
     }
 
-    pub(crate) fn all_types_with_move_constructors(
-        &self,
-    ) -> impl Iterator<Item = QualifiedName> + '_ {
-        self.all_names()
-            .filter(|qn| self.get(qn).unwrap().has_move_constructor)
-            .cloned()
-    }
-
-    pub(crate) fn all_types_with_const_copy_constructors(
-        &self,
-    ) -> impl Iterator<Item = QualifiedName> + '_ {
-        self.all_names()
-            .filter(|qn| self.get(qn).unwrap().has_const_copy_constructor)
-            .cloned()
+    pub(crate) fn has_move_constructor(&self, qn: &QualifiedName) -> bool {
+        self.get(qn).unwrap().has_move_constructor
```

Comment:
To avoid the unwrap, I'd prefer that you had a single API something like this:

```rust
struct KnownTypeConstructorDetails {
  has_move_constructor: bool,
  has_const_copy_constructor: bool,
}

pub(crate) fn get_known_constructor_details(&self, qn: &QualifiedName) -> Option<KnownTypeConstructorDetails>
```

Reply:
Done